### DISCUSSION TITLE
Paging in the DataGrid control

### DISCUSSION THREAD

**Discussion by FSharpCSharp** [AvaloniaUI/Avalonia#9579] [Q&A] [answered]
During my search for a functionality to use paging in the DataGrid I noticed that there are already TODO comments about this in the code in some places, among others in the DataGridCollectionView, there is very often the comment "TODO Paging".

What about the topic in general in the DataGrid? Especially for large amounts of data, paging would be really helpful in the standard DataGrid, also from the point of view of performance, etc. Can one of the product developers perhaps give me a statement on whether this is already in the backlog or planned? That would be really great!

Thanks a lot in advance!

**Comment by maxkatz6** [comment, marked as answer]
> whether this is already in the backlog or planned

No, and unlikely to be.
For two main reasons:
- we invest more time into TreeDataGrid control which has better integration with Avalonia
- features like datagrid paging or filtering can be relatively easily implemented on top of the control by developers

**Reply by timunie** [reply]
To add: `DynmaicData` has support for this, so you can easily integrate it into your ViewModel. 

Example: https://www.codeproject.com/Articles/5284771/Data-Paging-with-Dynamic-Data

**Reply by FSharpCSharp** [reply]
Best thanks to both of you for the quick replies. Was more or less just wondering about the TODO comments in the code. It makes more sense that way, WPF doesn't offer that by default either. Of course it makes much more sense this way! I find Avalonia really great and your work really noteworthy. This is really wonderful!
